### PR TITLE
Cvs 132196 dump to file

### PR DESCRIPTION
**Pull request by rasapala**

### CODE REVIEW

**Comment by dtrawins** [comment]
Add readme changes to describe how this feature is expected to be used.

File: mediapipe/calculators/ovms/openvinoinferencedumputils.cc [hunk 1]
```diff
@@ -0,0 +1,229 @@
+//*****************************************************************************
+// Copyright 2023 Intel Corporation
```
Comment (dtrawins) [inline mediapipe/calculators/ovms/openvinoinferencedumputils.cc]: wrong header date

File: mediapipe/calculators/ovms/openvinoinferencedumputils.cc [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+namespace mediapipe {
+    
+std::unordered_map<std::string, int> dumpCounters;
+
+using InferenceInput = std::map<std::string, ov::Tensor>;
+
+static std::stringstream dumpOvTensor(const ov::Tensor& tensor) {
+    std::stringstream dumpStream;
+    switch (tensor.get_element_type()) {
+        // The cosine window and square root of Hann are equivalent.
```
Comment (atobiszei) [inline mediapipe/calculators/ovms/openvinoinferencedumputils.cc]: ??

File: mediapipe/calculators/ovms/openvinoinferencedumputils.cc [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+        case ov::element::Type_t::f64: {
+            const _Float64* input_tensor_access = reinterpret_cast<_Float64*>(tensor.data());
+            dumpStream << " tensor: [ ";
+            for (int x = 0; x < tensor.get_size(); x++) {
+                dumpStream << &input_tensor_access[x] << " ";
+            }
+            dumpStream << " ]";
+            break;
+        }
```
Comment (atobiszei) [inline mediapipe/calculators/ovms/openvinoinferencedumputils.cc]: Why the `&`? Shouldn't it be `&input_tensor_access[x] --> input_tensor_access[x]`?
I would also just rename it to `value`.
```suggestion
#define TYPE_CASE(ENUM, TYPE) \
        case ENUM: { \
            const TYPE* input_tensor_access = reinterpret_cast<TYPE*>(tensor.data()); \
            dumpStream << " tensor: [ "; \
            for (int x = 0; x < tensor.get_size(); x++) { \
                dumpStream << &input_tensor_access[x] << " "; \
            } \
            dumpStream << " ]"; \
            break; \
        }
        TYPE_CASE(ov::element::Type_t::f64, _Float64*)
        TYPE_CASE(ov::element::Type_t::f32, _Float32*)
        ...
```
Then reuse for all.

File: mediapipe/calculators/ovms/modelapiovmsadapter.cc [hunk 4]
```diff
@@ -117,6 +119,9 @@ InferenceOutput OVMSInferenceAdapter::infer(const InferenceInput& input) {
             OVMS_BUFFERTYPE_CPU,
             NOT_USED_NUM));
     }
+#if (OVMS_DUMP_TO_FILE == 1)
```
Comment (atobiszei) [inline mediapipe/calculators/ovms/modelapiovmsadapter.cc:122]: This file will be moved at some point to model_api repository, so I am not sure if we should leave it here in addition to inference calculator. On the other hand for now it is the only way to debug geti graphs.

**Review by atobiszei — APPROVED** [review]

**Review by dtrawins — APPROVED** [review]

**Pull request merged**